R/length.R

```diff
@@ -121,35 +121,27 @@ xportr_length <- function(.df,
   # Check any variables missed in metadata but present in input data ---
   miss_vars <- setdiff(names(.df), metadata[[variable_name]])
 
-  miss_length <- as.character()
-  if (length_source == "metadata") {
+  miss_length <- character(0L)
+  width_attr <- if (identical(length_source, "metadata")) {
     length_metadata <- metadata[[variable_length]]
     names(length_metadata) <- metadata[[variable_name]]
 
     # Check any variables with missing length in metadata
     miss_length <- names(length_metadata[is.na(length_metadata)])
 
-    for (i in names(.df)) {
-      if (i %in% miss_vars) {
-        attr(.df[[i]], "width") <- length_data[[i]]
-      } else if (is.na(length_metadata[[i]])) {
-        attr(.df[[i]], "width") <- length_data[[i]]
-      } else {
-        attr(.df[[i]], "width") <- length_metadata[[i]]
-      }
-    }
-  }
-
-  # Message for missing var and missing length
-  length_log(miss_vars, miss_length, verbose)
-
-  # Assign length from data
-  if (length_source == "data") {
-    for (i in names(.df)) {
-      attr(.df[[i]], "width") <- length_data[[i]]
-    }
-
-
+    # Build `width` attribute
+    vapply(
+      names(.df),
+      function(i) {
+        if (i %in% miss_vars || is.na(length_metadata[[i]])) {
+          as.numeric(length_data[[i]])
+        } else {
+          as.numeric(length_metadata[[i]])
```

Comment:
@sophie-gem can you confirm that this can be safely coerced to numeric?

Thanks!

(avoids a `for` loop that adds complexity)

Reply:
Hi @averissimo, I didn't work on the `length.R` script - I think it may have been @cpiraux - can you help here?

Reply:
I confirm it can be coerced to numeric